### PR TITLE
chore: migrate dapp env vars to constants

### PR DESCRIPTION
**Pull request by nichosystem**
# Description
- fixed protocol local-deploy script
- moved dapp env vars to json constants
- removed some deprecated code from dapp

dependent on #433 

# Checklist
- [x] Code follows the style guide
- [x] I have performed a self-review of my own code
- [x] New and existing tests pass locally
- [x] This PR is targeting the correct branch

### CODE REVIEW

File: apps/dapp/src/providers/StakingProvider.tsx
Comment (nichosystem) [inline apps/dapp/src/providers/StakingProvider.tsx:1]: looks like my linter might have a different max length than what this file was previously linted as, sorry for the many changes 🙇

File: apps/dapp/src/services/MetricsService.ts
Comment (nichosystem) [inline apps/dapp/src/services/MetricsService.ts:1]: removing this nocheck revealed a ton of errors in this file, so quite a few general fixes besides migrating away from env vars

File: apps/dapp/src/services/MetricsService.ts
Comment (nichosystem) [inline apps/dapp/src/services/MetricsService.ts:176]: removed `signer` altogether and replaced with the static `Factory.connect(address, provider)` method, since everything should be readonly (we also get to ditch the env var SERVER_PRIVATE_KEY from this 👍 )

File: protocol/scripts/deploys/helpers.ts
Comment (nichosystem) [inline protocol/scripts/deploys/helpers.ts:76]: only actual change in this file is adding these 2 addresses, everything else is linting

**Review by mrfujisawa — APPROVED** [review]

File: apps/dapp/src/constants/env/local.tsx
Comment (butlerji) [inline apps/dapp/src/constants/env/local.tsx:7]: I'd expect this to no longer be used/required by the DApp (IMHO I think this should be tracked as an issue, and hopefully someone in the community can pick it up as a low priority/nice starter task).

File: apps/dapp/src/constants/env/local.tsx
Comment (butlerji) [inline apps/dapp/src/constants/env/local.tsx:8]: I'd expect this to no longer be used/required by the DApp (IMHO I think this should be tracked as an issue, and hopefully someone in the community can pick it up as a low priority/nice starter task).

File: apps/dapp/src/constants/env/local.tsx
Comment (butlerji) [inline apps/dapp/src/constants/env/local.tsx:9]: What's this used for?

File: apps/dapp/src/constants/env/local.tsx
Comment (butlerji) [inline apps/dapp/src/constants/env/local.tsx:13]: What's are these used for?

File: apps/dapp/src/constants/env/production.tsx
Comment (butlerji) [inline apps/dapp/src/constants/env/production.tsx:3]: confirming there are no secrets (and safe to checkin/have in git).

Guessing the dapp doesn't have secrets in general?

File: apps/dapp/src/constants/env/types.ts
Comment (butlerji) [inline apps/dapp/src/constants/env/types.ts:6]: Given we have a lovely mono repo @nichosystem - I think this should only live in one place.

I suggest deleting the list of active contracts from the protocol repo, and keeping this as the single source of truth, for active contracts WDYT?

I can leave the markdown of deprecated/deleted contracts in the protocol repo (that feels like the right home for it), but no need to keep 'active' contracts listed in 2 places

**Review by butlerji — APPROVED** [review]
LGTM

General comments are more discussion points/things to follow up on. Nothing blocking this merge :)

File: apps/dapp/src/constants/env/types.ts
Comment (butlerji) [reply apps/dapp/src/constants/env/types.ts:6]: Update, started a draft PR for this: https://github.com/TempleDAO/temple/pull/446

I'm not going to actually unify the contracts in this PR, Leave it to whoever does issue https://github.com/TempleDAO/temple/issues/435

File: apps/dapp/src/constants/env/production.tsx
Comment (nichosystem) [reply apps/dapp/src/constants/env/production.tsx:3]: yeah dapp doesnt have secrets

File: apps/dapp/src/constants/env/local.tsx
Comment (nichosystem) [reply apps/dapp/src/constants/env/local.tsx:13]: currently pulled into the metrics service to display the treasury. I think this work is migrated to the subgraph though, we can return to the metrics service cleanup in a future issue

File: apps/dapp/src/constants/env/local.tsx
Comment (nichosystem) [reply apps/dapp/src/constants/env/local.tsx:8]: I think we may want to wait a bit until there's some full consensus that faith is for sure dead

**Pull request merged**
